src/Agent.Worker/Variables.cs

```diff
@@ -407,7 +405,7 @@ public void Unset(string name)
             }
         }
 
-        public void Set(string name, string val, bool secret = false, bool readOnly = false)
+        public void Set(string name, string val, bool secret = false, bool readOnly = false, bool preserveCase = false)
```

Comment:
won't fixing this unless there's serious push back :)

keeping consistent with existing usage, and likely more risk updating all call sites here

Reply:
does this change the behavior?  I agree with the change, but it might be better to put it behind a knob/feature flag (off by default) if the behavior is changed.

Reply:
It does not change behavior by default. Users would need to opt-in to this behavior on a given variable by specifying this flag:

```shell
echo ##vso[task.setvariable variable=MyCasing;preservecase=true;]my value
```